src/main/java/graphql/execution/nextgen/depgraph/DependencyGraph.java

```diff
@@ -0,0 +1,123 @@
+package graphql.execution.nextgen.depgraph;
+
+import graphql.language.OperationDefinition;
+import graphql.schema.GraphQLSchema;
+import graphql.util.TraversalControl;
+import graphql.util.Traverser;
+import graphql.util.TraverserContext;
+import graphql.util.TraverserVisitorStub;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+
+public class DependencyGraph {
+
+
+    public DependencyGraph createDependencyGraph(GraphQLSchema graphQLSchema, OperationDefinition operationDefinition) {
+
+        FieldCollectorWTC fieldCollector = new FieldCollectorWTC();
+        FieldCollectorParameters parameters = FieldCollectorParameters
+                .newParameters()
+                .schema(graphQLSchema)
+                .build();
+
+        Function<MergedFieldWithType, List<MergedFieldWithType>> getChildren = mergedFieldWithType -> {
+            List<MergedFieldWithType> childs = fieldCollector.collectFields(parameters, mergedFieldWithType);
+            return childs;
+        };
+
+
+        Set<FieldVertex> allVertices = new LinkedHashSet<>();
+
+        Traverser<MergedFieldWithType> traverser = Traverser.depthFirst(getChildren);
+        FieldVertex rootVertex = new FieldVertex(null, null, null, null, null);
+        traverser.rootVar(FieldVertex.class, rootVertex);
+        allVertices.add(rootVertex);
+        List<MergedFieldWithType> roots = fieldCollector.collectFromOperation(parameters, operationDefinition, graphQLSchema.getQueryType());
+        traverser.traverse(roots, new TraverserVisitorStub<MergedFieldWithType>() {
+            @Override
+            public TraversalControl enter(TraverserContext<MergedFieldWithType> context) {
+                MergedFieldWithType mergedFieldWithType = context.thisNode();
+                System.out.println(mergedFieldWithType.getName() + "" + mergedFieldWithType.getObjectType().getName());
+
+                FieldVertex fieldVertex = createFieldVertex(mergedFieldWithType, graphQLSchema);
+                FieldVertex parentVertex = context.getVarFromParents(FieldVertex.class);
+                fieldVertex.addDependency(parentVertex);
+                parentVertex.addDependsOnMe(fieldVertex);
```

Comment:
Perhaps, it would be easier for the client if fieldVertex.addDependency had recorded dependencies both ways. Conceptually "dependency" represents an edge between two vertices, so in my mind it would be more convenient if all necessary bookkeeping was done in one operation.